examples/utils.py

```diff
@@ -99,18 +99,27 @@ def train_with_torch(
 
     for step, batch in enumerate(dataloader):
         inputs, labels = preproc(step, batch) if preproc is not None else batch
-        loss = model(*inputs, labels=labels).loss
+        # print(f"inputs from the batch: {inputs}")
+        # print(f"labels from the batch: {labels}")
+        if hasattr(model, 'loss_fn'):
+            # use customized loss_fn due to sharded linear layer
+            outputs = model(*inputs)
+            loss = model.loss_fn(outputs.logits, labels)
+        else:
+            loss = model(*inputs, labels=labels).loss
```

Comment:
This might be confusing. Maybe we could do the following:

```python
def train_with_torch(
    model, dataloader, optimizer=None, loss_fn=None, ...
):
    ...
    if loss_fn is not None:
        outputs = model(*inputs)
        loss = model.loss_fn(outputs.logits, labels)
    else:
        loss = model(*inputs, labels=labels).loss
```
Actually I could also change this in #31 so that we don't need to resolve conflicts between PRs.